### ISSUE TITLE
Preparing Custom Dataset for Training Model

### ISSUE THREAD

**Issue by talatccan** [milesial/Pytorch-UNet#169]
Hi,

I want to prepare my own dataset for train model but i do not know how to prepare it.
Dataset have grayscale x-ray images with jpeg format. Images sizes are 2943x1435.
Also there are json files which are contains coordinates of masks for each image and I can create mask images by drawing those json files.

What details should I pay attention to when drawing masks? Masks colors should be same as example output?

**Comment by milesial** [comment]
Hi. Yes you should create image masks. If you have 2 classes, coding them as black & white is fine. You can save your mask as you want as long as you modify the preprocess method to match your needs https://github.com/milesial/Pytorch-UNet/blob/d66f2c3d35c1c7dc58c6d35c6ee6aa05cacfbccf/utils/dataset.py#L26

**Comment by talatccan** [comment]
Thank you for reply. I did some modify on dataset.py and it worked.

**Comment by milesial** [comment]
Yes play more with the learning rate it an important factor in convergence
